exec/java-exec/src/main/java/org/apache/drill/exec/physical/rowSet/impl/TupleSetImpl.java

```diff
@@ -0,0 +1,551 @@
+import org.apache.drill.exec.expr.TypeHelper;
+import org.apache.drill.exec.physical.rowSet.ColumnLoader;
+import org.apache.drill.exec.physical.rowSet.TupleLoader;
+import org.apache.drill.exec.physical.rowSet.TupleSchema;
+import org.apache.drill.exec.physical.rowSet.impl.ResultSetLoaderImpl.VectorContainerBuilder;
+import org.apache.drill.exec.record.BatchSchema;
+import org.apache.drill.exec.record.BatchSchema.SelectionVectorMode;
+import org.apache.drill.exec.record.MaterializedField;
+import org.apache.drill.exec.vector.AllocationHelper;
+import org.apache.drill.exec.vector.ValueVector;
+import org.apache.drill.exec.vector.VectorOverflowException;
+import org.apache.drill.exec.vector.accessor.impl.AbstractColumnWriter;
+import org.apache.drill.exec.vector.accessor.impl.ColumnAccessorFactory;
+
+/**
+ * Implementation of a column when creating a row batch.
+ * Every column resides at an index, is defined by a schema,
+ * is backed by a value vector, and and is written to by a writer.
+ * Each column also tracks the schema version in which it was added
+ * to detect schema evolution. Each column has an optional overflow
+ * vector that holds overflow record values when a batch becomes
+ * full.
+ * <p>
+ * Overflow vectors require special consideration. The vector class itself
+ * must remain constant as it is bound to the writer. To handle overflow,
+ * the implementation must replace the buffer in the vector with a new
+ * one, saving the full vector to return as part of the final row batch.
+ * This puts the column in one of three states:
+ * <ul>
+ * <li>Normal: only one vector is of concern - the vector for the active
+ * row batch.</li>
+ * <li>Overflow: a write to a vector caused overflow. For all columns,
+ * the data buffer is shifted to a harvested vector, and a new, empty
+ * buffer is put into the active vector.</li>
+ * <li>Excess: a (small) column received values for the row that will
+ * overflow due to a later column. When overflow occurs, the excess
+ * column value, from the overflow record, resides in the active
+ * vector. It must be shifted from the active vector into the new
+ * overflow buffer.
+ */
+
+public class TupleSetImpl implements TupleSchema {
+
+  public static class TupleLoaderImpl implements TupleLoader {
+
+    public TupleSetImpl tupleSet;
+
+    public TupleLoaderImpl(TupleSetImpl tupleSet) {
+      this.tupleSet = tupleSet;
+    }
+
+    @Override
+    public TupleSchema schema() { return tupleSet; }
+
+    @Override
+    public ColumnLoader column(int colIndex) {
+      // TODO: Cache loaders here
+      return tupleSet.columnImpl(colIndex).writer;
+    }
+
+    @Override
+    public ColumnLoader column(String colName) {
+      ColumnImpl col = tupleSet.columnImpl(colName);
+      if (col == null) {
+        throw new UndefinedColumnException(colName);
+      }
+      return col.writer;
+    }
+
+    @Override
+    public TupleLoader loadRow(Object... values) {
```

Comment:
Type validation is done as the values wend their way through the writer tree. Eventually, we'll notice that the value is, say, Integer and call the `setInt()` method on the writer. If the writer is really for, say, a VarChar, then an unsupported operation exception will be thrown at that time. Similarly, if the object type is not one we know how to parse, then an exception will be thrown at that time.